legal-api/src/legal_api/services/filings/validations/registration.py

```diff
@@ -0,0 +1,131 @@
+# Copyright © 2022 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validation for the Registration filing."""
+from datetime import timedelta
+from http import HTTPStatus  # pylint: disable=wrong-import-order
+from typing import Dict, Final, Optional
+
+import pycountry
+from dateutil.relativedelta import relativedelta
+from flask_babel import _ as babel  # noqa: N813, I004, I001, I003
+from legal_api.errors import Error
+from legal_api.models import Business, PartyRole
+from legal_api.utils.legislation_datetime import LegislationDatetime
+
+from ...utils import get_date, get_str
+from .common_validations import validate_court_order
+
+
+def validate(registration_json: Dict) -> Optional[Error]:
+    """Validate the Registration filing."""
+    if not registration_json:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid filing is required.')}])
+    legal_type = get_str(registration_json, '/filing/business/legalType')
+    msg = []
+
+    msg.extend(validate_business_type(registration_json, legal_type))
+    msg.extend(validate_party(registration_json, legal_type))
+    msg.extend(validate_start_date(registration_json))
+    msg.extend(validate_delivery_address(registration_json))
```

Comment:
Shouldnt there be a name request validation?

Reply:
To verify if its can be consumed? I did not find that in IA validation. I was referring to that since both are creating a new business.

Reply:
Idk why it is missing from IA. But it has to be validated. Things that needs to get validated are 
1. whether it can be consumed
2. Legal type is the same

Reply:
we should probably create a ticket for IA processor to make sure this is done too.